nvflare/fuel/f3/streaming/stream_utils.py

```diff
@@ -39,3 +39,54 @@ def gen_stream_id():
     with lock:
         stream_count += 1
     return f"SID{(start_time + stream_count):16.0f}"
+
+
+class FastBuffer:
+    """A buffer with fast appending"""
+
+    def __init__(self, buf: BytesAlike = None):
+        if not buf:
+            self.capacity = 1024
+        else:
+            self.capacity = len(buf)
+
+        self.buffer = bytearray(self.capacity)
+        if buf:
+            self.buffer[:] = buf
+            self.size = len(buf)
+        else:
+            self.size = 0
+
+    def to_bytes(self) -> BytesAlike:
+        """Return bytes-like object.
+        Once this method is called, append() may not work any longer, since the buffer may have been exported"""
+
+        if self.capacity == self.size:
+            result = self.buffer
+        else:
+            view = wrap_view(self.buffer)
+            result = view[0 : self.size]
+
+        return result
+
+    def append(self, buf: BytesAlike):
+        """Fast append by doubling the size of the buffer when it runs out"""
+
+        if not buf:
+            return self
+
+        length = len(buf)
+        remaining = self.capacity - self.size
+        if length > remaining:
+            # Expanding the array as least twice the current capacity
+            new_cap = max(length + self.size, 2 * self.capacity)
+            self.buffer.ljust(new_cap, b"\x00")
+            self.capacity = new_cap
+
+        self.buffer[self.size :] = buf
+        self.size += length
+
+        return self
```

Comment:
do we need a method to shrink back the buffer size? This is to handle the case if the system temporary hit a scenario need to expand the buffer capacity for a particular application case. Once the application quiets down, the system can release the un-needed allocated memory.

Reply:
Probably not. All these buffers are temporary. They should all be freed when it's done.